cmd/api/src/api/v2/cypher_search_integration_test.go

```diff
@@ -85,3 +86,47 @@ func Test_CypherSearch(t *testing.T) {
 		}),
 	)
 }
+
+func Test_CypherSearch_WithoutCypherMutationsEnabled(t *testing.T) {
+	harness := lab.NewHarness()
+
+	// Default ConfigFixture does not have cypher mutations enabled
+	adminApiClientFixture := fixtures.NewAdminApiClientFixture(fixtures.ConfigFixture, fixtures.NewApiFixture())
+	lab.Pack(harness, adminApiClientFixture)
+
+	lab.NewSpec(t, harness).Run(
+		lab.TestCase("errors on mutations with enable cypher_mutations false", func(assert *require.Assertions, harness *lab.Harness) {
+			apiClient, ok := lab.Unpack(harness, adminApiClientFixture)
+			assert.True(ok)
+
+			_, err := apiClient.CypherSearch(v2.CypherSearch{Query: "match (w) where w.name = 'voldemort' remove w.name return w"})
+			assert.ErrorContains(err, frontend.ErrUpdateClauseNotSupported.Error())
+		}),
+	)
+}
+
+func Test_CypherSearch_WithCypherMutationsEnabled(t *testing.T) {
+	customCfg, err := utils.LoadIntegrationTestConfig()
+	require.Nil(t, err)
+
+	// Enable cypher mutations in API
+	customCfg.EnableCypherMutations = true
+
+	harness := lab.NewHarness()
+	customCfgFixture := fixtures.NewCustomConfigFixture(customCfg)
+	lab.Pack(harness, customCfgFixture)
+	customApiFixture := fixtures.NewCustomApiFixture(customCfgFixture)
+	lab.Pack(harness, customApiFixture)
+	adminApiClientFixture := fixtures.NewAdminApiClientFixture(customCfgFixture, customApiFixture)
+	lab.Pack(harness, adminApiClientFixture)
+
+	lab.NewSpec(t, harness).Run(
+		lab.TestCase("errors on mutations with enable cypher_mutations true", func(assert *require.Assertions, harness *lab.Harness) {
```

Comment:
It looks like this description should say something like, "does not throw errors with cypher_mutations enabled" since currently it reads like it is supposed to error even though we are asserting a nil err.

Reply:
I'd even go shorter: "Mutations allowed with enable_cypher_mutations true"